src/main/kotlin/app/revanced/patches/grindr/unlimited/patch/setUnlimited.kt

```diff
@@ -0,0 +1,125 @@
+            return v0
+        """.trimIndent()
+
+        val _false = """
+            const/4 v0, 0x0
+            return v0
+        """.trimIndent()
+
+        /*
+            Based on: https://github.com/ElJaviLuki/GrindrPlus
+         */
+
+        HasFeatureFingerprint.result?.let { result ->
+            println("Found HasFeatureFingerprint!")
+            result.mutableMethod.apply {
+                replaceInstructions(0, _true)
+            }
+        } ?: throw HasFeatureFingerprint.exception
+
+        IsFreeFingerprint.result?.let { result ->
+            println("Found IsFreeFingerprint!")
+            result.mutableMethod.apply {
+                addInstruction(3, """
+                    const/4 v0, 0x0
+                """.trimIndent())
+            }
+        } ?: throw IsFreeFingerprint.exception
+
+        IsNoPlusUpsellFingerprint.result?.let { result ->
+            println("Found IsNoPlusUpsellFingerprint!")
+            result.mutableMethod.apply {
+                replaceInstructions(0, _true)
+            }
+        } ?: throw IsNoPlusUpsellFingerprint.exception
+
+        IsNoXtraUpsellFingerprint.result?.let { result ->
+            println("Found IsNoXtraUpsellFingerprint!")
+            result.mutableMethod.apply {
+                replaceInstructions(0, _true)
+            }
+        } ?: throw IsNoXtraUpsellFingerprint.exception
+
+        IsPlusFingerprint.result?.let { result ->
+            println("Found IsPlusFingerprint!")
+            result.mutableMethod.apply {
+                replaceInstructions(0, _false)
+            }
+        } ?: throw IsPlusFingerprint.exception
+
+        IsUnlimitedFingerprint.result?.let { result ->
+            println("Found IsUnlimitedFingerprint!")
+            result.mutableMethod.apply {
+                replaceInstructions(0, _true)
+            }
+        } ?: throw IsUnlimitedFingerprint.exception
+
+        IsXtraFingerprint.result?.let { result ->
+            println("Found IsXtraFingerprint!")
+            result.mutableMethod.apply {
+                replaceInstructions(0, _false)
+            }
+        } ?: throw IsXtraFingerprint.exception
+
+        //this must always be true
+        InnaccessibleProfileManagerbFingerprint.result?.let { result ->
+            println("Found InnaccessibleProfileManagerbFingerprint!")
+            result.mutableMethod.apply {
+                replaceInstructions(0, _true)
+            }
+        } ?: throw InnaccessibleProfileManagerbFingerprint.exception
+
+        //this must always be false (the opposite of InnaccessibleProfileManagerbFingerprint)
+        InnaccessibleProfileManagerdFingerprint.result?.let { result ->
+            println("Found InnaccessibleProfileManagerdFingerprint!")
+            result.mutableMethod.apply {
+                addInstruction(3, """
+                    const/4 v0, 0x0
+                """.trimIndent())
+            }
+        } ?: throw InnaccessibleProfileManagerdFingerprint.exception
```

Comment:
This is a duplicate code. Create a map between fingerprints and the modification code and loop over it to apply the patch.